Updating to most recent version of libnfs

tryggvil opened the pull request:
- Updating to most recent version of libnfs master
- Updating nix crate
- Changing Rc to Arc
- Removing problematic redeclaration of nfs_url struct

tryggvil committed:
Updating to most recent version of libnfs

tryggvil committed:
Updating temporarily to local libnfs-sys version

cholcombe973 commented on Cargo.toml:
```diff
@@ -12,8 +12,11 @@ keywords = ["nfs"]
 libc = "~0.2"
-libnfs-sys = "~0.2"
-nix = "~0.25"
+#libnfs-sys = "~0.2"
```
We'll need to change this to the published version again. I'll push that change to crates.io shortly

cholcombe973 commented on src/lib.rs:
```diff
@@ -3,24 +3,25 @@
 #[derive(Clone)]
 struct NfsPtr(*mut nfs_context);
 
+unsafe impl Send for NfsPtr{}
```
I think back when I first wrote this I tried to figure out if the nfs pointer is thread safe. Could you point me towards some code or docs?

tryggvil replied:
I updated the implementation to enclose nfs_context within a Mutex to implement safe Send

cholcombe973 commented:
I found my old issue I opened with LibNFS: https://github.com/sahlberg/libnfs/issues/266. It's not thread safe so we shouldn't impl Send/Sync

tryggvil committed:
Updating implementation to use Mutex

fridvin-netapp committed:
Changed implementation of pread_into and pwrite

... so that reads/writes happen in chunks where maximum chunk size comes from max
read/write size obtained via libnfs functions nfs_get_readmax/nfs_get_writemax,
respectively.

fridvin-netapp committed:
Updated libnfs-sys dependency

Now references our forked repo and its version.
